presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/BenchmarkSTIntersects.java

```diff
@@ -0,0 +1,163 @@
+    {
+        return stIntersects(data.crossingLine, data.simpleGeometry);
+    }
+
+    @Benchmark
+    public Object stIntersectsOuterLineInEnvelope(BenchmarkData data)
+    {
+        return stIntersects(data.outerLineInEnvelope, data.geometry);
+    }
+
+    @Benchmark
+    public Object stIntersectsOuterLineInEnvelopeSimpleGeometry(BenchmarkData data)
+    {
+        return stIntersects(data.outerLineInEnvelope, data.simpleGeometry);
+    }
+
+    @Benchmark
+    public Object stIntersectsOuterLineNotInEnvelope(BenchmarkData data)
+    {
+        return stIntersects(data.outerLineNotInEnvelope, data.geometry);
+    }
+
+    @Benchmark
+    public Object stIntersectsOuterLineNotInEnvelopeSimpleGeometry(BenchmarkData data)
+    {
+        return stIntersects(data.outerLineNotInEnvelope, data.simpleGeometry);
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private Slice simpleGeometry;
+        private Slice geometry;
+        private Slice innerLine;
+        private Slice crossingLine;
+        private Slice outerLineInEnvelope;
+        private Slice outerLineNotInEnvelope;
+
+        @Setup
+        public void setup()
+                throws IOException
+        {
+            simpleGeometry = stGeometryFromText(utf8Slice("POLYGON ((16.5 54, 16.5 54.1, 16.51 54.1, 16.8 54))"));
+            geometry = stGeometryFromText(utf8Slice(loadPolygon("large_polygon.txt")));
+            innerLine = stGeometryFromText(utf8Slice("LINESTRING (16.6 54.0167, 16.6 54.017)"));
+            crossingLine = stGeometryFromText(utf8Slice("LINESTRING (16.6 53, 16.6 56)"));
+            outerLineInEnvelope = stGeometryFromText(utf8Slice("LINESTRING (16.6667 54.05, 16.8667 54.05)"));
+            outerLineNotInEnvelope = stGeometryFromText(utf8Slice("LINESTRING (16.6667 54.25, 16.8667 54.25)"));
+        }
+
+        public void validate()
+        {
+            validate(simpleGeometry);
+            validate(geometry);
+        }
+
+        public void validate(Slice geometry)
+        {
+            Slice envelope = stEnvelope(geometry);
+
+            // innerLine
+            verify(stIntersects(geometry, innerLine));
+            verify(stContains(geometry, innerLine));
+
+            // crossingLine
+            verify(stIntersects(geometry, crossingLine));
+            verify(!stContains(geometry, crossingLine));
+
+            // outerLineInEnvelope
+            verify(!stIntersects(geometry, outerLineInEnvelope));
+            verify(stIntersects(envelope, outerLineInEnvelope));
+
+            // outerLineNotInEnvelope
+            verify(!stIntersects(geometry, outerLineNotInEnvelope));
+            verify(!stIntersects(envelope, outerLineNotInEnvelope));
+        }
+    }
+
+    public static void main(String[] args)
+            throws IOException, RunnerException
+    {
+        BenchmarkData data = new BenchmarkData();
+        data.setup();
+        data.validate();
```

Comment:
nit: @martint Martin told me a trick of adding @Test method with validation logic. This ensures that benchmark code stays valid even if benchmark is not run.

Reply:
Turns out that files with a "Benchmark*" prefix weren't picked by SureFire plugin for execution as TestNG tests. I had to add yet another commit [https://github.com/prestodb/presto/pull/10182/commits/b4845a6fcddd54f78f6566dd9534f15b5e535f22]

Reply:
@electrum